### PR TITLE
APP-31283 -- added domain detection for integration key

### PR DESCRIPTION
**Pull request by awright96**
This change is [Reviewable](https://reviewable.io/reviews/pendo-io/zendo/1)

### CODE REVIEW

File: src/sources/PendoClient.js [hunk 1]
```diff
@@ -44,7 +44,20 @@ const Pendo = {
     // R.map((key) => R.partial(Pendo[key], token), R.reject(R.equals('initialize'),R.keys(Pendo)))
   },
 
+  setUrl (token) {
```
Comment (brec-c) [inline src/sources/PendoClient.js]: You already mentioned not handling the `.us` correct but it also looks like you're not updating the `token` to drop the `.<region>` suffix after updating the URL.

File: src/sources/PendoClient.js [hunk 2]
```diff
@@ -33,9 +33,9 @@ const toLowerify = (mdFieldName) => {
   }
 }
 
-const Pendo = {
+var Pendo = {
```
Comment (brec-c) [inline src/sources/PendoClient.js]: you should be able to use `const` here.

**Review by brec-c — COMMENTED** [review]
You might consider adding a new function that takes the incoming `token` and returns an object with the new `token` and `url` as properties.

**Comment by brec-c** [comment]
___
*[src/sources/PendoClient.js, line 69 at r2](https://reviewable.io/reviews/pendo-io/zendo/1#-M799GsQ2vPpKTnxAYOK:-M799GsR1aOHu3XkD76j:b-qx1vzd) ([raw file](https://github.com/pendo-io/zendo/blob/4a9371edf304537d5bd72aa4bcdb0462ed79d3b8/src/sources/PendoClient.js#L69)):*
> ```JavaScript
> 
>   fetchUserById (token, email) {
>     const [url, integrationKey] = this.getUrl(token);
> ```

just commented on this but I think it was in the wrong spot?

pretty sure this needs to be `const {url, integrationKey} = this.getUrl(token);` in order to destructure an object, not an array

**Review by brec-c — COMMENTED** [review]
Looking good. Either small syntax issue or I'm about to learn something new

> *[Reviewable](https://reviewable.io/reviews/pendo-io/zendo/1#-:-M799r9N0c0QSyK0o3oU:bveho7t)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @awright96)

**Review by awright96 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pendo-io/zendo/1)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @brec-c)

___
*[src/sources/PendoClient.js, line 69 at r2](https://reviewable.io/reviews/pendo-io/zendo/1#-M799GsQ2vPpKTnxAYOK:-M79I1Nf18vZL8c3hAl9:b-896fix) ([raw file](https://github.com/pendo-io/zendo/blob/4a9371edf304537d5bd72aa4bcdb0462ed79d3b8/src/sources/PendoClient.js#L69)):*
<details><summary><i>Previously, brec-c (Brec Carson) wrote…</i></summary><blockquote>

just commented on this but I think it was in the wrong spot?

pretty sure this needs to be `const {url, integrationKey} = this.getUrl(token);` in order to destructure an object, not an array
</blockquote></details>

Done.

**Review by brec-c — APPROVED** [review]
:lgtm:

> Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/pendo-io/zendo/1#-:-M79ML246EMMczXskshl:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by brec-c — APPROVED** [review]
[merge]

> *[Reviewable](https://reviewable.io/reviews/pendo-io/zendo/1#-:-M79MMiOEwziJ4BGR2tw:b3sq3yz)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**